### PR TITLE
Change GPU to OpenCL

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

There are a number of simple changes: 

- moved `stan/math/gpu` to `stan/math/opencl`
- moved `test/unit/math/gpu` to `test/unit/math/opencl`
- changed all references of `matrix_gpu` to `matrix_cl`
- changed the references `matrix on the GPU` to `OpenCL matrix`

## Tests

`/gpu` Tests were moved to `/opencl`

## Side Effects

Not sure on the Jenkins changes, hopefully I didnt break anything regarding the tests. Other changes are trivial.

## Checklist

- [x] Math issue #1133 

- [x] Copyright holder: Rok Češnovar, Univ. of Ljubljana

- [x] the basic tests are passing

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
The GPU tests fail because the files were moved to `/opencl` or because I changed the Jenkins file.

**Comment by SteveBronder** [comment]
Looks like some changed wasn't reflected on Jenkins. Tried to kick it off but we'll see. I'm running all the tests locally to check if something was missed

**Comment by SteveBronder** [comment]
@seantalts do you know why Jenkins is still referencing `./runTests.py -j16 test/unit/math/gpu --make-only`?

It looks like it's changed in the Jenkinsfile

File: Jenkinsfile [hunk 1]
```diff
@@ -159,16 +159,16 @@ pipeline {
                     }
                     post { always { retry(3) { deleteDir() } } }
                 }
-                stage('GPU Tests') {
-                    agent { label "gpu" }
+                stage('OpenCL Tests') {
+                    agent { label "opencl" }
```
Comment (seantalts) [inline Jenkinsfile]: we probably want to keep this label as 'GPU' because it specifies that we want a node with a GPU

**Comment by seantalts** [comment]
You can more easily kick off just a specific part of a Jenkins job by going to the job, making sure you're logged in, and then hitting the button to restart just that phase:
image

That way we don't get a bunch of garbage commits in our git history, too :)

**Comment by seantalts** [comment]
I thiiiink this might be an artifact of a security policy that basically doesn't trust new Jenkinsfiles from forks by default... Can we put this onto a branch on `stan-dev` and try it? I think you should both be members of the stan-dev org and writers to math already, no?

**Comment by SteveBronder** [comment]
Yeah that makes sense I'll make a new branch in stan-dev

**Pull request closed without merging**